src/main/java/org/sefglobal/scholarx/service/MentorService.java

```diff
@@ -15,9 +19,37 @@
 
     private final static Logger log = LoggerFactory.getLogger(MentorService.class);
     private final MentorRepository mentorRepository;
+    private final ProgramRepository programRepository;
+    private final ProfileRepository profileRepository;
 
-    public MentorService(MentorRepository mentorRepository) {
+    public MentorService(MentorRepository mentorRepository, ProgramRepository programRepository, ProfileRepository profileRepository) {
         this.mentorRepository = mentorRepository;
+        this.programRepository = programRepository;
+        this.profileRepository = profileRepository;
+    }
+
+    /**
+     * Create new {@link Mentor}
+     *
+     * @param profileId which is the profile id of the applying user's {@link Profile}
+     * @param programId which is the requested {@link Program}
+     * @return the created {@link Mentor}
+     * @throws ResourceNotFoundException is thrown if the applying {@link Program} doesn't exist
+     */
+    public Mentor applyAsMentor(long programId, long profileId) throws ResourceNotFoundException {
+        Optional<Program> optionalProgram = programRepository.findById(programId);
+        if (!optionalProgram.isPresent()) {
+            String msg = "Error, Unable to apply as a mentor. " +
+                         "Program with id: " + programId + " doesn't exist.";
+            log.error(msg);
+            throw new ResourceNotFoundException(msg);
+        }
+
+        Mentor mentor = new Mentor();
+        mentor.setProfile(profileRepository.getOne(profileId));
```

Comment:
`profileId` should be validated for the following scenario,
* Does the profile exist?

Reply:
Done